CMakeLists.txt

```diff
@@ -369,11 +369,16 @@ if(WITH_PROMETHEUS)
     if(EXISTS ${PROJECT_SOURCE_DIR}/third_party/prometheus-cpp/.git)
       set(SAVED_ENABLE_TESTING ${ENABLE_TESTING})
       set(SAVED_CMAKE_CXX_CLANG_TIDY ${CMAKE_CXX_CLANG_TIDY})
+      set(SAVED_CMAKE_CXX_INCLUDE_WHAT_YOU_USE
```

Comment:
Does prometheus overwrite the value of `CMAKE_CXX_INCLUDE_WHAT_YOU_USE`?

Reply:
No, but the way CMake works, we need to clear it before entering third_party/prometheus, otherwise the IWYU build will also report issues on the prometheus header themselves, when CMake builds the prometheus library.